clientside/src/features/board/components/Ship.tsx

```diff
@@ -0,0 +1,118 @@
+import { FC } from "react";
+import styled from "styled-components";
+import {
+    BattleshipDirection,
+    BattleshipPartType,
+    BattleshipType,
+} from "../types/battleship";
+
+export interface Props {
+    size?: number;
+    color?: string;
+    part: BattleshipPartType;
+    direction: BattleshipDirection;
+    type?: BattleshipType;
+}
+
+const ShipPart: FC<Props> = ({
+    part,
+    direction,
+    size = 32,
+    color = "#a9d5fd",
+    type = BattleshipType.Default,
+}) => {
+    let rotateDeg = 0;
+    switch (direction) {
```

Comment:
I think if this could be a dictionary would be nice so you could do stuff like `rotateDeg[BattleshipDirection.Horizontal]` or something like that